### PR TITLE
New-DbaAgentSchedule does not create schedule for multiple instances where there is a job provided + test

### PR DESCRIPTION
**Pull request by MikeyBronowski**
## Type of Change

 - [x] Bug fix (non-breaking change, fixes # )
 - [ ] New feature (non-breaking change, adds functionality, fixes # )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
 
The fix is in addition to latest PR #6644 
Added the test to test both scenarios
### Commands to test

```
$server1 = "localhost:14333"
$server2 = "localhost:1433"
# create schedules on multiple instances - without jobs
New-DbaAgentSchedule -SqlInstance $server1,$server2 -Schedule dbatoolsci_ScheduleWithoutJobOnMultipleServers -Force

# all good, both instances reporting the schedule was created
Get-DbaAgentSchedule -SqlInstance $server1 -Schedule dbatoolsci_ScheduleWithoutJobOnMultipleServers | FT
Get-DbaAgentSchedule -SqlInstance $server2 -Schedule dbatoolsci_ScheduleWithoutJobOnMultipleServers | FT

# create jobs without a schedule on multiple instances
New-DbaAgentJob -SqlInstance $server1,$server2 -Job dbatoolsci_JobWithoutScheduleAtFirst
Get-DbaAgentJob -SqlInstance $server1,$server2 -Job dbatoolsci_JobWithoutScheduleAtFirst | FT

# create schedules for those jobs on all instances
New-DbaAgentSchedule -SqlInstance $server1,$server2 -Job dbatoolsci_JobWithoutScheduleAtFirst -Schedule dbatoolsci_NewScheduleForAJobWithoutASchedule -Force

# the is reporting the schedule was created on one instance only
Get-DbaAgentSchedule -SqlInstance $server1 -Schedule dbatoolsci_NewScheduleForAJobWithoutASchedule | FT
Get-DbaAgentSchedule -SqlInstance $server2 -Schedule dbatoolsci_NewScheduleForAJobWithoutASchedule | FT

# cleanup
Remove-DbaAgentJob -SqlInstance $server1,$server2 -Job dbatoolsci_JobWithoutScheduleAtFirst
Remove-DbaAgentSchedule -SqlInstance $server1,$server2 -Schedule dbatoolsci_NewScheduleForAJobWithoutASchedule, dbatoolsci_ScheduleWithoutJobOnMultipleServers
```

### CODE REVIEW

**Comment by potatoqualitee** [comment]
Looks like there was a failure in the specific test that I'll have to take a closer look at. I think there are conflicting parents.

File: functions/New-DbaAgentSchedule.ps1 [hunk 1]
```diff
@@ -534,7 +534,7 @@ function New-DbaAgentSchedule {
                             }
 
                             # Output the job schedule
-                            return $JobSchedule
+                            $JobSchedule
                         }
                     }
                 } # foreach object job
```
Comment (wsmelton) [inline functions/New-DbaAgentSchedule.ps1:540]: Can you remove the comment on this line and push to the PR to get the test to run again, please.

Seems Pester hit an exception for the test on this command for some reason.

```Exception calling "LeaveTestGroup" with "2" argument(s): "TestGroups stack corrupted:  Expected name/hint of 'C:\github\dbatools\tests\New-DbaAgentSchedule.Tests.ps1','Script'.  Found 'New-DbaAgentSchedule Integration Tests', 
'Describe'."
At C:\Program Files\WindowsPowerShell\Modules\Pester\4.4.2\Pester.psm1:880 char:21
+ ...                    $pester.LeaveTestGroup($testScript.Path, 'Script')
+                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : ScriptMethodRuntimeException
 
Running C:\github\dbatools\tests\New-DbaAzAccessToken.Tests.ps1
Running C:\github\dbatools\tests\New-DbaClientAlias.Tests.ps1
```

Comment (MikeyBronowski) [reply functions/New-DbaAgentSchedule.ps1:540]: Interesting. Will make the change

Comment (MikeyBronowski) [reply functions/New-DbaAgentSchedule.ps1:539]: @wsmelton not sure if that comment you've meant (line 536), but the test failed again.

Comment (wsmelton) [reply functions/New-DbaAgentSchedule.ps1:539]: Line 540

**Comment by wsmelton** [comment]
The test indeed is failing because of the change made as other PRs do not show the test on this command failing.

Please just remove the test update from this PR and we will address it when we being reworking all of our testing framework.

**Pull request merged**